### PR TITLE
updates for Microsoft.DotNet.Interactive.Documents API stabilization

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs [hunk 1]
```diff
@@ -264,19 +259,23 @@ internal static void MergeKernelInfos(KernelInfoCollection destination, KernelIn
...
 
-    internal static bool TryGetKernelInfoFromMetadata(
+    public bool TryGetKernelInfosFromMetadata(
+        [NotNullWhen(true)] out KernelInfoCollection? kernelInfos) =>
+        TryGetKernelInfosFromMetadata(Metadata, out kernelInfos);
+
+    internal static bool TryGetKernelInfosFromMetadata(
         IDictionary<string, object>? metadata,
-        [NotNullWhen(true)] out KernelInfoCollection? kernelInfo)
+        [NotNullWhen(true)] out KernelInfoCollection? kernelInfos)
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs:268]: Can this be inlined into the above non-static overload (with all callers switched to calling the non-static overload)? Or can this be made private?

Please ignore the above if we have more callers for the static overload outside the current class.

File: src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ public async IAsyncEnumerable<InteractiveDocument> GetImportsAsync(bool recursiv
     {
         EnsureImportFieldParserIsInitialized();
 
-        if (!TryGetKernelInfoFromMetadata(Metadata, out var kernelInfos))
+        if (!TryGetKernelInfosFromMetadata(out var kernelInfos))
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs:68]: minor nit: This call uses the public instance overload while others below use the static overload. Could we standardize to the same call?

File: src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs [hunk 3]
```diff
@@ -198,7 +198,7 @@ internal void NormalizeElementKernelNames(KernelInfoCollection kernelInfos)
 
     public string? GetDefaultKernelName()
     {
-        if (TryGetKernelInfoFromMetadata(Metadata, out var kernelInfo))
+        if (TryGetKernelInfosFromMetadata(Metadata, out var kernelInfo))
         {
             return kernelInfo.DefaultKernelName;
         }
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs:204]: Why is the public API missing the remaining heuristics (like checking for` kernelspec` metadata)

**Review by shyamnamboodiripad — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs [hunk 4]
```diff
@@ -264,19 +259,23 @@ internal static void MergeKernelInfos(KernelInfoCollection destination, KernelIn
         destination.AddRange(source.Where(ki => added.Add(ki.Name)));
     }
 
-    internal static bool TryGetKernelInfoFromMetadata(
+    public bool TryGetKernelInfosFromMetadata(
```
[the remark this answers is not in the document]
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs:262]: We discussed this during the sync, that there's currently no way to get the kernel info using the public API, and that it's very useful to have.

**Pull request merged**